umbraco/Umbraco-CMS
# Fixing #6441 so empty media items are not detected as folders.

Fixing #6441 so empty media items are not detected as folders.

Few things involved here:

* hack implemented in EntityController for media items to ensure that an IsFolder metadata value is set based on whether the media type has an umbracoFile property type (basically the previously logic from < 7.15 for folder detection)
* when resolving media items from the entity controller (these are 'slim' media items), always pass this data through mediaHelper.formatMediaEntityData to ensure that isFolder and image/thumbnail properties are correctly set. Now we don't have to do this logic differently in all the places. This method has a few fallback mechanisms for backwards compat, etc... 

Testing: 

* See #6441 
* Create a media item file but don't upload a file, then try to pick this in a picker with images only = true
* Ensure all media pickers work correctly with all the different types of options
* Ensure all media grid views work correctly
